### PR TITLE
re: bug-commit-analysis files

### PR DESCRIPTION
**Pull request by EYH0602**

### CODE REVIEW

File: src/format.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import csv
```
Comment (EYH0602) [inline src/format.py:1]: what is the purpose of this file?

File: src/count.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            time.sleep(2 ** attempt) 
+             # Failed to fetch data after retries
+    return None 
+
+def read_repos(file_path):
+    with open(file_path, 'r') as file:
+        repos = [line.strip() for line in file.readlines()]
+    return repos
+
+def get_commits_count(repo):
```
Comment (EYH0602) [inline src/count.py:40]: please document the return value of this function, why `str | int`?

File: src/count.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            return last_page_number
+        elif response.json():  
+            # Check if there's at least one commit in the specified range
+            return 1
+            # No commits in the specified date range
+        return 0  
+    return "Error"
+
+
+def get_bugs_count(repo):
```
Comment (EYH0602) [inline src/count.py:58]: why return type is `str | int`, I suggest `Optional[int]` or `Maybe[int]` (see docs for [Maybe](https://returns.readthedocs.io/en/latest/pages/maybe.html))

File: src/count.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+
+
+def write_output(repos_info, output_file):
+    with open(output_file, 'w') as file:
+        for repo, commits_count, bugs_count in repos_info:
+            file.write(f"{repo.replace('/', ',')} , {commits_count} , {bugs_count}\n")
+
+def main():
+    if len(sys.argv) < 2:
```
Comment (EYH0602) [inline src/count.py:86]: use [`fire`](https://google.github.io/python-fire/guide/) might be easier

File: src/count.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        print("Usage: python script.py input_file_path")
+        sys.exit(1)
+
+    file_path = sys.argv[1]
+    output_file = 'output4.csv'
```
Comment (EYH0602) [inline src/count.py:91]: don't hardcode this if possible

File: src/format.py (hunk 1 above)
Comment (acharyadarshan) [reply src/format.py:1]: my original output file wasn't in the structure I wanted it to be. So I am just organizing my repo/id, commit count, and bug count in the format I want similar to what you have in google drive

Comment (EYH0602) [reply src/format.py:1]: Modifying `src/count.py` to write csv is more straightforward. We only want to keep the process of reproducing our results here.

**Pull request closed without merging**